hail/python/hail/genetics/pedigree.py

```diff
@@ -201,8 +200,40 @@ def read(cls, fam_path, delimiter='\\s+') -> 'Pedigree':
         :rtype: :class:`.Pedigree`
         """
 
-        jrep = Env.hail().methods.Pedigree.read(fam_path, Env.hc()._jhc.sFS(), delimiter)
-        return Pedigree._from_java(jrep)
+        trios = []
+        missing_sex_count = 0
+        missing_sex_values = set()
+        with Env.fs().open(fam_path) as file:
+            for line in file:
+                split_line = re.split(delimiter, line.strip())
```

Comment:
I would have written `line.strip().split(delimiter)`. I'm not sure which is better. Your call.

Reply:
In this case, I am pretty sure I had to use `re` because the default delimiter pattern is written as a regex expression to match all different kinds of whitespace. If I recall correctly, I originally had what you specified and it didn't work with the `\s+` pattern.